src/dmd/typesem.d

```diff
@@ -746,13 +746,13 @@ extern(C++) Type typeSemantic(Type type, const ref Loc loc, Scope* sc)
         if (mtype.dim)
         {
             //https://issues.dlang.org/show_bug.cgi?id=15478
+            //https://issues.dlang.org/show_bug.cgi?id=21870
             if (mtype.dim.isDotVarExp())
             {
                 if (Declaration vd = mtype.dim.isDotVarExp().var)
                 {
-                    FuncDeclaration fd = vd.toAlias().isFuncDeclaration();
-                    if (fd)
-                        mtype.dim = new CallExp(loc, fd, null);
+                    if (FuncDeclaration fd = vd.toAlias().isFuncDeclaration())
+                        mtype.dim = new CallExp(loc, mtype.dim);
```

Comment:
Shouldn't `fd` be passed to `CallExp`? If not, why is it declared?

Reply:
will remove the decl. it is not required anymore.

Reply:
to clarify the bug was that `a.b.fd` was rewritten  `fd()` so now instead `fd` is checked to be callable and the rewrite does not drop `a.b` as this may give a `this` (or more).